process/track/metaBlockTrack.go

```diff
@@ -91,8 +83,8 @@ func NewMetaBlockTrack(arguments ArgMetaTracker) (*metaBlockTrack, error) {
 	mbt.blockProcessor = blockProcessor
 
 	mbt.headers = make(map[uint32]map[uint64][]*headerInfo)
-	mbt.metaBlocksPool.RegisterHandler(mbt.receivedMetaBlock)
-	mbt.shardHeadersPool.RegisterHandler(mbt.receivedShardHeader)
+	mbt.headersPool.RegisterHandler(mbt.receivedMetaBlock)
```

Comment:
Now, because both methods are notified when a new shards/meta header is received we should put a filter in each method (in receivedMetaBlock to avoid headers with shardId <> MetaChain and in receivedShardHeader to avoid headers with shardId == MetaChain)

Reply:
done.